internal/types/kruizePayload/common.go

```diff
@@ -282,22 +291,15 @@ func make_container_data(c map[string]interface{}) container {
 	return container_data
 }
 
-func MemBytesToMi(mem float64) float64 {
-	memoryInMi := mem / 1024 / 1024
-	truncatedMem := math.Trunc(memoryInMi*100) / 100
-	return truncatedMem
-}
-
-func isMemoryField(metricName string) bool {
-	// TO DO: In go 1.21 use slices package instead of map
-	memoryFields := map[string]string{"memoryRSS": "", "memoryUsage": "", "memoryLimit": "", "memoryRequest": ""}
-	_, keyExists := memoryFields[metricName]
-	return keyExists
-}
-
-func convertMemoryFields(metricName string, field string, c map[string]interface{}) {
-	if isMemoryField(metricName) {
-		converted_mem := MemBytesToMi(c[field].(float64))
-		c[field] = converted_mem
+func convertMemoryFields(field string, c map[string]interface{}) {
```

Comment:
I have removed the Trunc usage from here as even if we Trunc the memory to 3 decimal points kruize returns it with more decimals.
At the same time if we don't update the c[field] in this function then we would end up repeating the below logic multiple times,
~~~
someField, ok := c[field].(float64)
if ok {
do the convert
}
else{
Warn
}
~~~

This is because go does not allow us to pass `c[field]` directly, it needs to be cast first and that would mean we cast it everytime before we pass it to func. So I kept the logic as is, and this also resembles to what we do at https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/api/utils.go#L231 only difference is here its separate func.

Reply:
Instead of setting c[field] = memoryInMi explicitly, we can capture the value of
convertMemoryFields function if it returns the same.

convertedValue := convertMemoryFields(max_field, c)

// Assign the max value returned
max = AssertAndConvertToString(convertedValue)

Reply:
Okay, I tried to give it a shot however then `AssertAndConvertToString` won't work for cpu and mem, it won't be generalized func? Or I will need to change float64->interface and then AssertAndConvertToString can handle it?

Reply:
At the moment `AssertAndConvertToString` seems to handle `float`, `int` and `string`, and looks like a general usage function